Update HR with 2020 change and add types

pero5ar opened the pull request:
In 2020 there has been a change in Holidays in Croatia, more info at: https://en.wikipedia.org/wiki/Public_holidays_in_Croatia - Updated data to reflect that change

Additionally added the types of the holidays along with the widely observed International Women's day .

Marked the regional holidays as optional because they don't cover the entire county (just the municipal areas of Split and Dubrovnik respectively) and do not necessarily include all types of business and services as public holidays do.

commenthol commented:
Hi @per5ar,

Thanks for contributing. I have some question regarding your pull request.

Labor Day shows up twice now in the list. 
One on May 1st `05-01` and another observed on March 8th `03-08`. The latter I can't find in the referenced source.

I also would like to maintain historic data, therefor I would propose the following changes. Would you please mind checking them and if all is fine pushing them in your PR?

```yaml
holidays:
  # @attrib https://en.wikipedia.org/wiki/Public_holidays_in_Croatia
  HR:
    names:
      hr: Hrvatska
      en: Croatia
    langs:
      - hr
    zones:
      - Europe/Belgrade
    days:
      01-01:
        _name: 01-01
        type: public
      01-06:
        _name: 01-06
        type: public
      easter -47:
        _name: easter -47
        type: observance
      easter:
        _name: easter
        type: public
      easter 1:
        _name: easter 1
        type: public
      easter 60:
        _name: easter 60
        type: public
      05-01:
        _name: 05-01
        type: public
      '05-30 #1':
        name:
          hr: Dan državnosti
          en: National Day
        type: observance
        active:
          - to: '2020-01-01'
      05-30:
        name:
          hr: Dan državnosti
          en: National Day
        type: public
        active:
          - from: '2020-01-01'
      2nd sunday in May:
        _name: Mothers Day
        type: observance
      06-22:
        name:
          hr: Dan antifašističke borbe
          en: Anti-Fascist Struggle Day
        type: public
      '06-25 #1':
        name:
          hr: Dan neovisnosti
          en: Statehood Day
        type: public
        active:
          - to: '2020-01-01'
      06-25:
        name:
          hr: Dan neovisnosti
          en: Statehood Day
        type: observance
        active:
          - from: '2020-01-01'
      08-05:
        name:
          hr: Dan pobjede i domovinske zahvalnosti
          en: Victory and Homeland Thanksgiving Day and the Day of Croatian defenders
        type: public
      08-15:
        _name: 08-15
        type: public
      '10-08 #1':
        name:
          hr: Dan Hrvatskoga sabora.
          en: Day of the Croatian Parliament
        type: public
        active:
          - to: '2020-01-01'
      10-08:
        name:
          hr: Dan Hrvatskoga sabora.
          en: Day of the Croatian Parliament
        type: observence
        active:
          - from: '2020-01-01'
      11-01:
        _name: 11-01
        type: public
      '11-18 #1':
        name:
          hr: Dan sjećanja na žrtve Domovinskog rata
          en: Remembrance Day
        type: observance
        active:
          - to: '2020-01-01'
      11-18:
        name:
          hr: Dan sjećanja na žrtve Domovinskog rata
          en: Remembrance Day
        type: public
        active:
          - from: '2020-01-01'
      12-25:
        _name: 12-25
        type: public
      12-26:
        _name: 12-26
        type: public
    regions:
      '17':
        name: Split-Dalmatia
        days:
          05-07:
            name:
              hr: Sveti Duje
              en: Saint Domnius
            type: optional
      '19':
        name: Dubrovnik-Neretva
        days:
          02-03:
            name:
              hr: Sveti Vlaho
              en: Saint Blaise
            type: optional
```

pero5ar commented:
I put the wrong name for `03-08`, It refers to [International Women's Day ](https://en.wikipedia.org/wiki/International_Women%27s_Day). It is not an official observance but something like a combination of Mother's Day, Valentine's Day, night marches for women rights, and is common on a per company basis for women to get an hour or so off on that date. It's probably the most noticeable non-public non-religious holiday in Croatia since Remembrance Day has been made public.

I've added the proposed changes, but without `'05-30 #1'` because there was no observance of anything on that date before (at least not since the '90s). It's basically the same holiday that was on `06-25` before, and the one on `06-25` now is the one that was on on `10-08` (the same name and purpose, the Parliament just changed which event they wanted to use for what). Yes, it's all unnecessarily confusing.

I've also corrected the `hr` names to match those of the The People's Newspaper (Croatia's official gazette): https://narodne-novine.nn.hr/clanci/sluzbeni/full/2019_11_110_2212.html 
There is unfortunately no official `en` posting so I'm treating the ones from Wikipedia as good enough.

This document also contains a list of observances some of which can be found on Wikipedia, but in practice most of them are only observed by some government bodies or special organisations related to them, so I'm not sure if they are worth adding.

One thing that is worth adding, but I'm not sure how to add, are holidays religious minorities?

commenthol commented:
For adding holidays for other religious groups you may take a look at BA.yaml.

pero5ar commented:
Added in a similar manner to `RS.yaml`. They seem to have the same law, just inverse for Orthodox and Catholic.

commenthol commented:
Thanks for your contribution
